robot_calibration/src/checkerboard_finder.cpp

```diff
@@ -172,11 +172,41 @@ bool CheckerboardFinder::findInternal(robot_calibration_msgs::CalibrationData *
     sensor_msgs::PointCloud2Iterator<float> iter_cloud(cloud_, "x");
 
     // Set msg size
-    msg->observations.resize(2);
-    msg->observations[0].sensor_name = camera_sensor_name_;
-    msg->observations[0].features.resize(points_x_ * points_y_);
-    msg->observations[1].sensor_name = chain_sensor_name_;
-    msg->observations[1].features.resize(points_x_ * points_y_);
+    // Handle multiple finders
+
+    int idx_cam = -1;
+    int idx_chain = -1;
+
+    if (msg->observations.size() == 0)
```

Comment:
we probably don't need this block -- each finder should be the exclusive creator for each sensor, right? thus, we should always just append two new observations